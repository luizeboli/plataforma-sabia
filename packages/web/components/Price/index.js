import React from 'react';
import PropTypes from 'prop-types';

const currencySimbol = {
	BRL: 'R$',
	USD: '$',
};

const Price = ({ amount, currency }) => {
	return amount ? (
		<span>
			{`${currencySimbol[currency]} ${new Intl.NumberFormat([], {
				currency,
				minimumFractionDigits: 2,
			}).format(amount)}`}
		</span>
	) : null;
};

Price.propTypes = {
	amount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
	currency: PropTypes.string,
};

Price.defaultProps = {
	currency: 'BRL',
};

export default Price;
