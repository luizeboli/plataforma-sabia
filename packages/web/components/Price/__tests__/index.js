import React from 'react';
import { render } from 'test-utils';
import Price from '..';

test('it render the Price component', () => {
	const { container } = render(<Price amount={5000} />);

	expect(container).toMatchSnapshot();
});

test('it render the Price component with the correct currency', () => {
	const { container } = render(<Price amount={5000} currency="USD" />);

	expect(container).toMatchSnapshot();
});
