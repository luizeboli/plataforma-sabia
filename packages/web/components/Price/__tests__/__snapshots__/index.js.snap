// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render the Price component 1`] = `
<div>
  <span>
    R$ 5,000.00
  </span>
</div>
`;

exports[`it render the Price component with the correct currency 1`] = `
<div>
  <span>
    $ 5,000.00
  </span>
</div>
`;
